sdk/cosmosdb/cosmos/src/diagnostics/Diagnostics.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+const _startTime = new Date().getTime();
+export const defaultHeaders: DiagnosticHeaders = {
+  cosmosdiagnostics: "Started Cosmos Diagnostics",
+  diagnosticStartTime: new Date().toLocaleString(),
+  durationInMs: new Date().getTime() - _startTime,
+};
+const _diagnosticHeaders: DiagnosticHeaders[] = [defaultHeaders];
+/**
+ * @internal
+ */
+export function setDiagnostics(message: DiagnosticHeaders | string) {
+  if (_diagnosticHeaders.length === 0) {
+    setDiagnostics(_diagnosticHeaders);
```

Comment:
should this be defaultHeaders? Otherwise, it will be endless recursion?

Reply:
done